### ISSUE TITLE
transform.ipynb not working [BUG]

### ISSUE THREAD

**Issue by koldunovn** [xgcm/xgcm#248]
**What happened**:

I am trying to run ``transform.ipynb`` and very early get:

```
AttributeError: 'Grid' object has no attribute 'transform'
```

Looks like the functionality wasn't merged yet, but the notebook is already available. Since it's not part of the 0.4.0 version that is under review, I guess it's fine, but it's a bit inconsisten to have it now in the ``doc`` while the functionality is not there yet :)
 
**What you expected to happen**:

Notebook runs without errors.

**xgcm version and environment**:

0.4.0 , but I am at commit e0eae1e

<details>
```
appnope                   0.1.0           py36h9f0ad1d_1001    conda-forge
argon2-cffi               20.1.0           py36h9de38fb_1    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     20.2.0             pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
beautifulsoup4            4.9.2                    pypi_0    pypi
bleach                    3.2.1              pyh9f0ad1d_0    conda-forge
bokeh                     2.2.1            py36h9f0ad1d_0    conda-forge
boost-cpp                 1.74.0               he5d75e3_0    conda-forge
brotlipy                  0.7.0           py36h9de38fb_1000    conda-forge
bzip2                     1.0.8                haf1e3a3_3    conda-forge
c-ares                    1.16.1               haf1e3a3_3    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
cairo                     1.16.0            ha8983da_1005    conda-forge
cartopy                   0.18.0           py36heb902f8_2    conda-forge
certifi                   2020.6.20        py36h9f0ad1d_0    conda-forge
cffi                      1.14.3           py36h6a9c9ac_0    conda-forge
cfitsio                   3.470                hdf94aef_6    conda-forge
cftime                    1.2.1                    pypi_0    pypi
chardet                   3.0.4           py36h9f0ad1d_1007    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
cloudpickle               1.6.0                      py_0    conda-forge
contextvars               2.4                        py_0    conda-forge
cryptography              3.1.1            py36h54f4bc5_0    conda-forge
curl                      7.71.1               hcb81553_6    conda-forge
cycler                    0.10.0                     py_2    conda-forge
cytoolz                   0.11.0           py36h9de38fb_0    conda-forge
dask                      2.28.0                     py_0    conda-forge
dask-core                 2.28.0                     py_0    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
distributed               2.28.0           py36h9f0ad1d_0    conda-forge
docopt                    0.6.2                    pypi_0    pypi
docrep                    0.2.7                      py_0    conda-forge
entrypoints               0.3             py36h9f0ad1d_1001    conda-forge
expat                     2.2.9                hb1e8313_2    conda-forge
fontconfig                2.13.1            h79c0d67_1002    conda-forge
freetype                  2.10.2               h8da9a1a_0    conda-forge
freexl                    1.0.5             h0b31af3_1002    conda-forge
fsspec                    0.8.3                      py_0    conda-forge
future                    0.18.2           py36h9f0ad1d_1    conda-forge
gdal                      3.1.2            py36h060f065_1    conda-forge
geos                      3.8.1                h4a8c4bd_0    conda-forge
geotiff                   1.6.0                h4e9c399_1    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
giflib                    5.2.1                h0b31af3_2    conda-forge
glib                      2.66.0               hdb5fb44_0    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.6          nompi_haae91d6_101    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
icu                       67.1                 hb1e8313_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
immutables                0.14             py36h37b9a7d_0    conda-forge
importlib-metadata        2.0.0            py36h9f0ad1d_0    conda-forge
importlib_metadata        2.0.0                         0    conda-forge
ipykernel                 5.3.4            py36h95af2a2_0    conda-forge
ipython                   5.8.0                    py36_1    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.5.1              pyh9f0ad1d_1    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
jpeg                      9d                   h0b31af3_0    conda-forge
json-c                    0.13.1            h575e443_1002    conda-forge
jsonschema                3.2.0            py36h9f0ad1d_1    conda-forge
jupyter                   1.0.0                      py_2    conda-forge
jupyter_client            6.1.7                      py_0    conda-forge
jupyter_console           5.2.0                    py36_1    conda-forge
jupyter_core              4.6.3            py36h9f0ad1d_1    conda-forge
jupyterlab_pygments       0.1.1              pyh9f0ad1d_0    conda-forge
kealib                    1.4.13               h40102fb_1    conda-forge
kiwisolver                1.2.0            py36h863e41a_0    conda-forge
krb5                      1.17.1               h75d18d8_3    conda-forge
lcms2                     2.11                 h174193d_0    conda-forge
libblas                   3.8.0               17_openblas    conda-forge
libcblas                  3.8.0               17_openblas    conda-forge
libclang                  10.0.1          default_hf57f61e_1    conda-forge
libcurl                   7.71.1               h9bf37e3_6    conda-forge
libcxx                    10.0.1               h5f48129_0    conda-forge
libdap4                   3.20.6               h993cace_1    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.2.1             hb1e8313_1007    conda-forge
libgdal                   3.1.2                h6dfbaa8_1    conda-forge
libgfortran               4.0.0               h2d743fc_11    conda-forge
libgfortran4              7.5.0               h2d743fc_11    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
libkml                    1.3.0             h8ca2c65_1012    conda-forge
liblapack                 3.8.0               17_openblas    conda-forge
libllvm10                 10.0.1               h009f743_3    conda-forge
libnetcdf                 4.7.4           nompi_hc5b2cf3_105    conda-forge
libnghttp2                1.41.0               h7580e61_2    conda-forge
libopenblas               0.3.10          openmp_h63d9170_4    conda-forge
libpng                    1.6.37               hb0a8c7a_2    conda-forge
libpq                     12.3                 h489d428_0    conda-forge
libsodium                 1.0.18               haf1e3a3_1    conda-forge
libspatialite             4.3.0a            h231dce8_1039    conda-forge
libssh2                   1.9.0                h8a08a2b_5    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libwebp-base              1.1.0                h0b31af3_3    conda-forge
libxml2                   2.9.10               h2c6e4a5_2    conda-forge
llvm-openmp               10.0.1               h28b9765_0    conda-forge
locket                    0.2.0                      py_2    conda-forge
lz4-c                     1.9.2                hb1e8313_3    conda-forge
markupsafe                1.1.1            py36h9de38fb_1    conda-forge
matplotlib-base           3.3.2            py36h534ab7b_0    conda-forge
mistune                   0.8.4           py36h37b9a7d_1001    conda-forge
msgpack-python            1.0.0            py36h863e41a_1    conda-forge
mysql-common              8.0.21                        2    conda-forge
mysql-libs                8.0.21               hfb8f7af_2    conda-forge
nbclient                  0.5.0                      py_0    conda-forge
nbconvert                 6.0.6            py36h9f0ad1d_0    conda-forge
nbformat                  5.0.7                      py_0    conda-forge
ncurses                   6.2                  hb1e8313_1    conda-forge
nest-asyncio              1.4.0                      py_1    conda-forge
netcdf4                   1.5.4                    pypi_0    pypi
notebook                  6.1.4            py36h9f0ad1d_0    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numpy                     1.19.1           py36h2fc57d0_2    conda-forge
olefile                   0.46                       py_0    conda-forge
openjpeg                  2.3.1                h254dc36_3    conda-forge
openssl                   1.1.1h               haf1e3a3_0    conda-forge
owslib                    0.20.0                     py_0    conda-forge
packaging                 20.4               pyh9f0ad1d_0    conda-forge
pandas                    1.1.2            py36h27176af_0    conda-forge
pandoc                    2.10.1               haf1e3a3_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
partd                     1.1.0                      py_0    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pexpect                   4.8.0            py36h9f0ad1d_1    conda-forge
pickleshare               0.7.5           py36h9f0ad1d_1001    conda-forge
pillow                    7.2.0            py36h2ae5dfa_1    conda-forge
pip                       20.2.3                     py_0    conda-forge
pixman                    0.38.0            h01d97ff_1003    conda-forge
poppler                   0.89.0               h3232a60_1    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                12.3                 h62ab893_0    conda-forge
proj                      7.1.0                h45baca5_1    conda-forge
prometheus_client         0.8.0              pyh9f0ad1d_0    conda-forge
prompt_toolkit            1.0.15                     py_1    conda-forge
psutil                    5.7.2            py36h9de38fb_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pydap                     3.2.2                    pypi_0    pypi
pyepsg                    0.4.0                      py_0    conda-forge
pygments                  2.7.1                      py_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.1.post1      py36hbb77884_1    conda-forge
pyqt                      5.12.3           py36haa9e2f4_3    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtchart                 5.12                     pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyrsistent                0.17.3           py36h9de38fb_0    conda-forge
pyshp                     2.1.2              pyh9f0ad1d_0    conda-forge
pysocks                   1.7.1            py36h9f0ad1d_1    conda-forge
python                    3.6.11          hc38f9c5_2_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.6                     1_cp36m    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyyaml                    5.3.1            py36h9de38fb_0    conda-forge
pyzmq                     19.0.2           py36hd529914_0    conda-forge
qt                        5.12.9               h717870c_0    conda-forge
qtconsole                 4.7.7              pyh9f0ad1d_0    conda-forge
qtpy                      1.9.0                      py_0    conda-forge
readline                  8.0                  h0678c8f_2    conda-forge
requests                  2.24.0             pyh9f0ad1d_0    conda-forge
scipy                     1.5.2            py36h01b1e2b_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                49.6.0           py36h9f0ad1d_1    conda-forge
shapely                   1.7.1            py36h1e0b1ed_0    conda-forge
simplegeneric             0.8.1                      py_1    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sortedcontainers          2.2.2              pyh9f0ad1d_0    conda-forge
soupsieve                 2.0.1                    pypi_0    pypi
sqlite                    3.33.0               h960bd1c_0    conda-forge
tbb                       2019.9               ha1b3eb9_1    conda-forge
tblib                     1.6.0                      py_0    conda-forge
terminado                 0.9.1            py36h9f0ad1d_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tiledb                    2.0.8                h8973ade_1    conda-forge
tk                        8.6.10               hb0a8c7a_0    conda-forge
toolz                     0.11.1                     py_0    conda-forge
tornado                   6.0.4            py36h37b9a7d_1    conda-forge
traitlets                 4.3.3            py36h9f0ad1d_1    conda-forge
typing_extensions         3.7.4.2                    py_0    conda-forge
tzcode                    2020a                h0b31af3_0    conda-forge
urllib3                   1.25.10                    py_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
webob                     1.8.6                    pypi_0    pypi
wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
widgetsnbextension        3.5.1            py36h9f0ad1d_1    conda-forge
xarray                    0.16.1                     py_0    conda-forge
xerces-c                  3.2.3                h0a0444a_1    conda-forge
xgcm                      0.4.0                    pypi_0    pypi
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                haf1e3a3_0    conda-forge
zeromq                    4.3.2                h4a8c4bd_3    conda-forge
zict                      2.0.0                      py_0    conda-forge
zipp                      3.2.0                      py_0    conda-forge
zlib                      1.2.11            h7795811_1009    conda-forge
zstd                      1.4.5                h289c70a_2    conda-forge
```

</details>

**Comment by rabernat** [comment]
Thanks for noting this @koldunovn. Looks like you are using the "latest" version of the docs, which is indeed our default:

https://xgcm.readthedocs.io/en/latest/

The "stable" version does not show unreleased features:

https://xgcm.readthedocs.io/en/stable/

We are in the process of making a new release this week. I'm sorry this caused confusion in your review.

**Comment by koldunovn** [comment]
@rabernat Thanks for clarification! Note, the link from the github repository actually lead to the "latest" version (default in rtd, I guess). 

But I get to the `transform.ipynb` just browsing throught the `doc` folder. It's a matter of taste, of course, but I would really make all your (excellent!) notebooks accesible from some separate folder, so it is easier for experienced user to begin using them straight away. 

I understand, though, that it might complicate building of documnetation.

**Comment by jbusecke** [comment]
I just released `v0.5.0`. It is updated on pypi, but not yet on conda-forge (usually takes a few hours).

**Comment by jbusecke** [comment]
This was automatically closed by #247, do you prefer to keep the issue open @koldunovn?
